src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordToString.cs

```diff
@@ -65,11 +65,11 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                 // builder.Append(" { ");
                 block.Add(makeAppendString(F, builderLocal, " { "));
 
-                // this.print(builder);
-                block.Add(F.ExpressionStatement(F.Call(F.This(), _printMethod, builderLocal)));
+                // if (this.print(builder)) builder.Append(" ");
```

Comment:
>// if (this.print(builder)) builder.Append(" "); [](start = 16, length = 48)

If `PrintMembers()` is only used internally, to implement `ToString()`, could this `if (this.print(builder))` be removed and simply have `PrintMembers()` add a " " before each member?

Reply:
> could this `if (this.print(builder))` be removed and simply have `PrintMembers()` add a " " before each member?

Given that `PrintMembers()` or `ToString()` can be implemented explicitly, it seems either requiring a trailing space here, or requiring a leading space before each member, will make one of the two methods somewhat awkward.

---
In reply to: [476183479](https://github.com/dotnet/roslyn/pull/47095#discussion_r476183479) [](ancestors = 476183479)

Reply:
I feel like the simpler implementation, for both us and consumers, is to just require a leading space in front of all members. Then there's no conditionals or questions.

---
In reply to: [476188140](https://github.com/dotnet/roslyn/pull/47095#discussion_r476188140) [](ancestors = 476188140,476183479)

Reply:
I think it's more regular to let the caller of `PrintMembers` to deal with the returned bool and using it to print the separators. We do that for printing the comma separator; the space separator should be no different.

---
In reply to: [476650024](https://github.com/dotnet/roslyn/pull/47095#discussion_r476650024) [](ancestors = 476650024,476188140,476183479)